### DISCUSSION TITLE
Output format JSON

### DISCUSSION THREAD

**Discussion by kuwv** [fluxcd/flux2#1904] [Proposals]
It would be nice if Flux CLI could output in JSON or other formats similar to `kubectl`.

**Comment by stefanprodan** [comment]
Flux can't reconcile manifests in JSON format only YAML, that's the reason `flux export` and `flux create --export` output is YAML only. Can you please say more on why you need JSON?

**Reply by kuwv** [reply]
It's much easier to process json with wrapper scripts than either yaml or a custom parser.

**Reply by stefanprodan** [reply]
Exporting Kubernetes manifests in a format that Flux does not support doesn't feel right to me. You could use `yq` instead of `jq` in your scripts or convert the YAML to JSON with some other tool.

**Reply by kuwv** [reply]
That may make sense for shell scripts, but that's not really practical for something like an Ansible module, etc.

But, if your not comfortable with it then don't sweat it.
